nautobot/extras/views.py

```diff
@@ -612,18 +613,53 @@ def extra_context(self):
         }
 
 
+def check_permission_and_workers(request, slug, func):
+    """Helper function for checking permissions, worker availability and running enqueue_pull_git_repository function
+
+    :param request:
+        request object
+
+    :param slug:
+        slug filed
+
+    :param func:
+        Enqueue git repo function
+
+
+    :return:
+        code 0: all good,
+        code 1: no worker process not running,
+        code 2: 403 Error
+    """
```

Comment:
Generally speaking we use [Google-style docstrings](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) in this project.

(I just checked and that's not explicitly called out in our style-guide documentation; we should rectify that.)